### ISSUE TITLE
Suppress "Unable to resolve configuration with compilerPath" warning in CMake+Swift projects

### ISSUE THREAD

**Issue by tristanlabelle** [microsoft/vscode-cmake-tools#4097]
### Environment

- OS and Version: Windows 11 22H2
- VS Code Version: 1.93.0
- C/C++ Extension Version: v1.21.6
- If using SSH remote, specify OS of remote machine: N/A

### Bug Summary and Steps to Reproduce

Bug Summary:
While working on a CMake Swift project, the C++ extension will regularly print the message below to the `C/C++ Configuration Warnings` output window righ after a build, which is annoying because:
1. It steals the focus from the `CMake/Build` output pane, hiding the output of the build
1. I'm not trying to use C/C++ code at the time where this shows up (though the project includes some C sources)
1. The C/C++ extension probably shouldn't be trying to resolve anything with the `swiftc.exe` compiler

```
[9/13/2024, 6:32:16 AM] Unable to resolve configuration with compilerPath "c:/users/tristan/appdata/local/programs/swift/toolchains/0.0.0+asserts/usr/bin/swiftc.exe".  Using "cl.exe" instead.
[9/13/2024, 6:36:45 AM] Unable to resolve configuration with compilerPath "c:/users/tristan/appdata/local/programs/swift/toolchains/0.0.0+asserts/usr/bin/swiftc.exe".  Using "cl.exe" instead.
```

Steps to reproduce:
1. Install Swift for Windows from https://www.swift.org/install/windows/
3. Clone https://github.com/tristanlabelle/swift-winrt
4. Open the `Generator` subfolder in VS Code from a VS dev cmd
5. Run `cmake --preset debug` and `cmake --build --preset debug`
(the bug is not specific to this repo, I see it regularly when working in Swift)

Expected behavior:
The C/C++ extension should not produce this warning. Probably it should ignore anything to do with the Swift compiler.

### Configuration and Logs

```shell
-------- Diagnostics - 9/13/2024, 6:53:56 AM
Version: 1.21.6
Current Configuration:
{
    "name": "Win32",
    "includePath": [
        "d:/swift-winrt/Generator/**"
    ],
    "defines": [
        "_DEBUG",
        "UNICODE",
        "_UNICODE"
    ],
    "windowsSdkVersion": "10.0.22621.0",
    "compilerPath": "cl.exe",
    "cStandard": "c17",
    "cppStandard": "c++17",
    "intelliSenseMode": "windows-msvc-x64",
    "compilerPathInCppPropertiesJson": "cl.exe",
    "intelliSenseModeIsExplicit": false,
    "cStandardIsExplicit": false,
    "cppStandardIsExplicit": false,
    "mergeConfigurations": false,
    "compilerPathIsExplicit": false,
    "configurationProvider": "ms-vscode.cmake-tools",
    "browse": {
        "path": [
            "d:/swift-winrt/Generator/**",
            "${workspaceFolder}"
        ],
        "limitSymbolsToIncludedHeaders": true
    }
}
Custom browse configuration: 
{
    "browsePath": [
        "d:/swift-winrt/generator/build/debug/dependencies/swift-argument-parser",
        "d:/swift-winrt/generator/build/debug/_deps/swift-argument-parser-src/sources/argumentparser/completions",
        "d:/swift-winrt/generator/build/debug/_deps/swift-argument-parser-src/sources/argumentparser/parsable properties",
        "d:/swift-winrt/generator/build/debug/_deps/swift-argument-parser-src/sources/argumentparser/parsable types",
        "d:/swift-winrt/generator/build/debug/_deps/swift-argument-parser-src/sources/argumentparser/parsing",
        "d:/swift-winrt/generator/build/debug/_deps/swift-argument-parser-src/sources/argumentparser/usage",
        "d:/swift-winrt/generator/build/debug/_deps/swift-argument-parser-src/sources/argumentparser/utilities",
        "d:/swift-winrt/generator/build/debug/_deps/swift-argument-parser-src/sources/argumentparsertoolinfo",
        "d:/swift-winrt/generator/sources/codewriters/c",
        "d:/swift-winrt/generator/sources/codewriters/cmake",
        "d:/swift-winrt/generator/sources/codewriters",
        "d:/swift-winrt/generator/sources/codewriters/swift",
        "d:/swift-winrt/generator/sources/codewriters/swift/syntaxwriters",
        "d:/swift-winrt/generator/build/debug/dependencies/swift-collections",
        "d:/swift-winrt/generator/build/debug/_deps/swift-collections-src/sources/collections",
        "d:/swift-winrt/generator/build/debug/_deps/swift-collections-src/sources/dequemodule",
        "d:/swift-winrt/generator/build/debug/dependencies/swift-dotnetmetadata",
        "d:/swift-winrt/generator/build/debug/_deps/swift-dotnetmetadata-src/sources/dotnetmetadatacinterop/include",
        "d:/swift-winrt/generator/build/debug/_deps/swift-dotnetmetadata-src/sources/dotnetmetadata",
        "d:/swift-winrt/generator/build/debug/_deps/swift-dotnetmetadata-src/sources/dotnetmetadata/attributes",
        "d:/swift-winrt/generator/build/debug/_deps/swift-dotnetmetadata-src/sources/dotnetmetadatacinterop",
        "d:/swift-winrt/generator/build/debug/_deps/swift-dotnetmetadata-src/sources/dotnetmetadataformat",
        "d:/swift-winrt/generator/build/debug/_deps/swift-dotnetmetadata-src/sources/dotnetxmldocs",
        "d:/swift-winrt/generator/build/debug/_deps/swift-collections-src/sources/orderedcollections/hashtable",
        "d:/swift-winrt/generator/build/debug/_deps/swift-collections-src/sources/orderedcollections/ordereddictionary",
        "d:/swift-winrt/generator/build/debug/_deps/swift-collections-src/sources/orderedcollections/orderedset",
        "d:/swift-winrt/generator/build/debug/_deps/swift-collections-src/sources/orderedcollections/utilities",
        "d:/swift-winrt/generator/build/debug/sources/codewriters",
        "d:/swift-winrt/generator/sources/projectionmodel",
        "d:/swift-winrt/generator/sources/projectionmodel/cabi",
        "d:/swift-winrt/generator/build/debug/sources/projectionmodel",
        "d:/swift-winrt/generator/sources/swiftwinrt",
        "d:/swift-winrt/generator/sources/swiftwinrt/writing",
        "d:/swift-winrt/generator/build/debug/sources/swiftwinrt",
        "d:/swift-winrt/generator/build/debug/_deps/swift-dotnetmetadata-src/sources/windowsmetadata/attributes",
        "d:/swift-winrt/generator/build/debug/_deps/swift-dotnetmetadata-src/sources/windowsmetadata"
    ],
    "compilerPath": "c:/users/tristan/appdata/local/programs/swift/toolchains/0.0.0+asserts/usr/bin/swiftc.exe",
    "compilerArgs": [],
    "compilerFragments": [
        "-sdk C:/Users/tristan/AppData/Local/Programs/Swift/Platforms/0.0.0/Windows.platform/Developer/SDKs/Windows.sdk/ -O -g -wmo -libc MD",
        "-suppress-remarks",
        "-suppress-warnings"
    ]
}
cpptools version (native): 1.21.6.0
Translation Unit Mappings:
[ D:\swift-winrt\Generator\.build\checkouts\swift-dotnetmetadata\Sources\DotNetMetadataCInterop\shim.c - source TU]:
Translation Unit Configurations:
[ D:\swift-winrt\Generator\.build\checkouts\swift-dotnetmetadata\Sources\DotNetMetadataCInterop\shim.c ]:
    Process ID: 18168
    Memory Usage: 50 MB
    Compiler Path: C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.39.33519\bin\Hostx64\x64\cl.exe
    Includes:
        C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.39.33519\include
        C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.39.33519\atlmfc\include
        C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Auxiliary\VS\include
        C:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\Include\um
    System Includes:
        C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\um
        C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt
        C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared
        C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\winrt
        C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\cppwinrt
    Defines:
        _DEBUG
        UNICODE
        _UNICODE
    Standard Version: ms_c17
    IntelliSense Mode: windows-msvc-x64
    Other Flags:
        --using_directory
        C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.39.33519\ATLMFC\lib\x64
        --using_directory
        C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.39.33519\lib\x64
        --using_directory
        C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.39.33519\lib\x86\store\references
        --using_directory
        C:\Program Files (x86)\Windows Kits\10\UnionMetadata\10.0.22621.0
        --using_directory
        C:\Program Files (x86)\Windows Kits\10\References\10.0.22621.0
        --using_directory
        C:\Windows\Microsoft.NET\Framework64\v4.0.30319
Total Memory Usage: 50 MB

------- Workspace parsing diagnostics -------
Number of files discovered (not excluded): 6004
```
```


### Other Extensions

CMake Tools and Swift extension, both of which I need (and more than the C/C++ extension in this scenario)

### Additional context

_No response_

**Comment by v-frankwang** [comment]
@tristanlabelle When we use the reproduction steps you provided to reproduce the issue, we get this error, can you give some advice?

![image](https://github.com/user-attachments/assets/7c033bd9-db86-4c83-9387-01e5e0d4f44f)

**Comment by tristanlabelle** [comment]
@v-frankwang , it looks like you don't have Swift installed as in my repro steps. `where.exe swiftc.exe` in the terminal should print the path to the swiftc compiler

**Comment by v-frankwang** [comment]
@tristanlabelle I still get the error after installing the Swift compiler.
[Error message.txt](https://github.com/user-attachments/files/17178736/Error.message.txt)

**Comment by tristanlabelle** [comment]
@v-frankwang please make sure to open VS Code from a VS Developer Command prompt which reports x64 for both host and target:
```
C:\Code>set VSCMD_ARG
VSCMD_ARG_HOST_ARCH=x64
VSCMD_ARG_TGT_ARCH=x64
```
You might want to use the explicit `x64 Native Tools Command Prompt for VS 2022` shortcut in the start menu.

**Comment by gcampbell-msft** [comment]
@tristanlabelle For the fix that @bobbrow pointed out, is the filegroup language for swift simply "swift"?

**Comment by tristanlabelle** [comment]
@gcampbell-msft , I don't know what that `filegroup` concept is. It doesn't seem to come from CMake, from a code search, so I assume it comes from the CMake Tools extension?

**Comment by gcampbell-msft** [comment]
@tristanlabelle I'll investigate, thanks.

**Comment by gcampbell-msft** [comment]
@tristanlabelle Could you test out the fix in this vsix? 

[cmake-tools.zip](https://github.com/user-attachments/files/17210918/cmake-tools.zip)

You'll have to change the file extension from .zip to .vsix after downloading.

**Comment by tristanlabelle** [comment]
@gcampbell-msft Sorry for the wait. Yup! That fixed it with my repro. Mind @'ing me in the PR?

**Comment by Yingzi1234** [comment]
This issue has been fixed on CMake Tools version 1.20.50(pre-release)

![Image](https://github.com/user-attachments/assets/d5b5b0b1-7d99-41c6-b8d1-16558739fd5b)
